src/rpc/handlers/AMMInfo.cpp

```diff
@@ -0,0 +1,207 @@
+    response.amount1 = toBoostJson(asset1Balance.getJson(JsonOptions::none));
+    response.amount2 = toBoostJson(asset2Balance.getJson(JsonOptions::none));
+    response.lpToken = toBoostJson(lptAMMBalance.getJson(JsonOptions::none));
+    response.tradingFee = to_string(amm[sfTradingFee]);
+    response.ammAccount = to_string(accID);
+
+    if (amm.isFieldPresent(sfVoteSlots)) {
+        for (auto const& voteEntry : amm.getFieldArray(sfVoteSlots)) {
+            boost::json::object vote;
+            vote[JS(account)] = to_string(voteEntry.getAccountID(sfAccount));
+            vote[JS(trading_fee)] = voteEntry[sfTradingFee];
+            vote[JS(vote_weight)] = voteEntry[sfVoteWeight];
+
+            response.voteSlots.push_back(std::move(vote));
+        }
+    }
+
+    if (amm.isFieldPresent(sfAuctionSlot)) {
+        auto const& auctionSlot = amm.peekAtField(sfAuctionSlot).downcast<STObject>();
+        if (auctionSlot.isFieldPresent(sfAccount)) {
+            boost::json::object auction;
+            auto const timeSlot = ammAuctionTimeSlot(lgrInfo.parentCloseTime.time_since_epoch().count(), auctionSlot);
+
+            auction[JS(time_interval)] = timeSlot ? *timeSlot : AUCTION_SLOT_TIME_INTERVALS;
+            auction[JS(price)] = toBoostJson(auctionSlot[sfPrice].getJson(JsonOptions::none));
+            auction[JS(discounted_fee)] = auctionSlot[sfDiscountedFee];
+            auction[JS(account)] = to_string(auctionSlot.getAccountID(sfAccount));
+            auction[JS(expiration)] = toIso8601(NetClock::time_point{NetClock::duration{auctionSlot[sfExpiration]}});
+
+            if (auctionSlot.isFieldPresent(sfAuthAccounts)) {
+                boost::json::array auth;
+                for (auto const& acct : auctionSlot.getFieldArray(sfAuthAccounts)) {
+                    boost::json::object accountData;
+                    accountData[JS(account)] = to_string(acct.getAccountID(sfAccount));
+                    auth.push_back(accountData);
+                }
+
+                auction[JS(auth_accounts)] = std::move(auth);
+            }
+
+            response.auctionSlot = std::move(auction);
+        }
+    }
+
+    if (!isXRP(asset1Balance)) {
+        response.asset1Frozen =
+            isFrozen(*sharedPtrBackend_, lgrInfo.seq, accID, input.issue1.currency, input.issue1.account, ctx.yield);
+    }
+    if (!isXRP(asset2Balance)) {
+        response.asset2Frozen =
+            isFrozen(*sharedPtrBackend_, lgrInfo.seq, accID, input.issue2.currency, input.issue2.account, ctx.yield);
+    }
+
+    response.ammID = to_string(ammKeylet.key);
+    return response;
+}
+
+void
+tag_invoke(boost::json::value_from_tag, boost::json::value& jv, AMMInfoHandler::Output const& output)
+{
+    boost::json::object amm = {
+        {JS(lp_token), output.lpToken},
+        {JS(amount), output.amount1},
+        {JS(amount2), output.amount2},
+        {JS(account), output.ammAccount},
+        {JS(trading_fee), output.tradingFee},
```

Comment:
please make sure the trading_fee type, should be number not string

Reply:
fixed